streams/src/main/java/org/apache/kafka/streams/errors/ProductionExceptionHandler.java

```diff
@@ -34,6 +34,18 @@ public interface ProductionExceptionHandler extends Configurable {
     ProductionExceptionHandlerResponse handle(final ProducerRecord<byte[], byte[]> record,
                                               final Exception exception);
 
+    /**
+     * Handles serialization exception and determine if the process should continue. The default implementation is to
+     * fail the process.
+     *
+     * @param record        the record that failed to serialize
+     * @param exception     the exception that occurred during serialization
+     */
+    default ProductionExceptionHandlerResponse onSerializationException(final ProducerRecord record,
+                                                                        final Exception exception) {
```

Comment:
maybe the exception should just be SerializationException

Reply:
I didn't see this before, but I agree, given it's exclusively for serialization exceptions.